### PR TITLE
Fix resource create/update

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves  resource create/update operation

## Changes
- added read helper method

### CODE REVIEW

File: twingate/client_resource.go [hunk 1]
```diff
@@ -21,6 +21,46 @@ type Resource struct {
...
+	}
+	RemoteNetwork struct {
+		ID graphql.ID
+	}
+	Groups struct {
+		PageInfo struct {
+			HasNextPage graphql.Boolean
+		}
+		Edges []*Edges
+	} `graphql:"groups(first: $first)"`
```
Comment (ekampf) [inline twingate/client_resource.go:38]: `$first` is not a good name as its very generic - what if we had multiple different connections we wanted to paginate here?
This should be something like `$groupsFirst` or `$groupsPageSize`

File: twingate/client_resource.go [hunk 2]
```diff
@@ -145,56 +185,40 @@ func convertPorts(ports []string) ([]*PortRangeInput, error) {
...
 
-func (client *Client) createResource(ctx context.Context, resource *Resource) error {
+func (client *Client) createResource(ctx context.Context, resource *Resource) (*Resource, error) {
 	variables := map[string]interface{}{
 		"name":            resource.Name,
 		"address":         resource.Address,
 		"remoteNetworkId": resource.RemoteNetworkID,
 		"groupIds":        resource.GroupsIds,
 		"protocols":       resource.Protocols,
+		"first":           graphql.Int(readResourceQueryGroupsSize),
```
Comment (ekampf) [inline twingate/client_resource.go:199]: See comment above - https://github.com/Twingate/terraform-provider-twingate/pull/196/files#r996166727

File: twingate/client_resource.go [hunk 3]
```diff
@@ -259,32 +267,39 @@ func (client *Client) readResources(ctx context.Context) ([]*Edges, error) { //n
...
-func (client *Client) updateResource(ctx context.Context, resource *Resource) error {
+func (client *Client) updateResource(ctx context.Context, resource *Resource) (*Resource, error) {
 	variables := map[string]interface{}{
 		"id":              resource.ID,
 		"name":            resource.Name,
 		"address":         resource.Address,
 		"remoteNetworkId": resource.RemoteNetworkID,
 		"groupIds":        resource.GroupsIds,
 		"protocols":       resource.Protocols,
+		"first":           graphql.Int(readResourceQueryGroupsSize),
```
Comment (ekampf) [inline twingate/client_resource.go:284]: same

**Review by ekampf — CHANGES_REQUESTED** [review]
Looks good but require a small fix

File: twingate/client_resource.go (hunk 1 above)
Comment (vmanilo) [reply twingate/client_resource.go:38]: good point, thank you

Comment (vmanilo) [reply twingate/client_resource.go]: updated

**Review by ekampf — APPROVED** [review]

**Pull request merged**